src/api/include/migraphx/migraphx.hpp

```diff
@@ -757,6 +757,13 @@ struct module
         return instruction(op_ins, own{});
     }
 
+    instruction add_literal(const migraphx::shape& s, const char* buffer)
```

Comment:
Maybe it would be better to do:

```cpp
template<class T>
instruction add_literal(const migraphx::shape& s, const T* buffer)
```

And then do a `static_cast<const char*>` in the function.

Reply:
Good idea. but it requires reinterpret_cast instead of static_cast.